backend/de.metas.business/src/main/java/de/metas/requisition/callout/M_Requisition.java

```diff
@@ -58,4 +64,17 @@ public void onDocTypeChanged(final I_M_Requisition requisition)
 			requisition.setDocumentNo(documentNoInfo.getDocumentNo());
 		}
 	}
+
+	@CalloutMethod(columnNames = I_M_Requisition.COLUMNNAME_M_Warehouse_ID)
+	public void onWarehouseChanged(@NonNull final I_M_Requisition requisition)
+	{
+		//
+		// M_Warehouse_ID is a mandatory column
+		final Optional<WarehouseId> warehouseId = WarehouseId.optionalOfRepoId(requisition.getM_Warehouse_ID());
+		if (warehouseId.isPresent())
+		{
+			final I_M_Warehouse warehouse = Services.get(IWarehouseDAO.class).getById(warehouseId.get());
+			requisition.setWarehouse_Location_ID(warehouse.getC_BPartner_Location_ID());
+		}
+	}
```

Comment:
@mrie04  u could write this as follows:
```java
WarehouseId.optionalOfRepoId(requisition.getM_Warehouse_ID())
.map(warehouseDAO::getById)
.ifPresent(warehouse -> requisition.setWarehouse_Location_ID(warehouse.getC_BPartner_Location_ID()) );
```

also pls extract warehouseDAO as a class field